pkg/driver/nodeserver.go

```diff
@@ -156,6 +156,11 @@ func (ns *NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 			fmt.Sprintf("failed to mount path %s -> %s: %s", mountPath, targetPath, err))
 	}
 
+	// We need to ensure unix permissions if the file already exists
+	if err := os.Chmod(targetPath, 700); err != nil {
+		return nil, err
+	}
```

Comment:
I'm confused by this. The comment says "file" already exists, but targetPath seems to be the directory where we have just mounted the "volume" containing the key and certs.

Will you add some further comments explaining what targetPath and mountPath are and explaining why we set their permissions this way.

What also confuses me is that `targetPath` seems to be what I would call a `mount point`, i.e. the path  at which we mount the file or directory.
And `mountPath` here seems to be what I would call the `sourcePath`, assuming it's a bind mount.

Anyway, some more comments would help me understand.

Reply:
The reason for this being here is simply precautionary. If the target path was created by some other mechanism, we are going to be overriding it with this permissions.